pragma alias silently ignores extra arguments

### Description

This is probably reported already, but reporting just in case.

```nim
var foo {.exportc: "abc".} = 123
{.pragma: importc2, importc.}
var bar {.importc2: "abc".}: int
echo bar
```

### Nim Version

[8f728ac](https://github.com/nim-lang/Nim/commit/8f728ace353857b007c9c38d1d4de98bf431d5e2)

### Current Output

```text
@min.nim.c.o: In function `NimMainModule':
@min.nim.c:(.text+0x267): undefined reference to `bar'
```


### Expected Output

```text
123
```


### Possible Solution

_No response_

### Additional Information

_No response_

--- a/compiler/pragmas.nim
+++ b/compiler/pragmas.nim
@@ -826,7 +826,8 @@ proc singlePragma(c: PContext, sym: PSym, n: PNode, i: var int,
                   validPragmas: TSpecialWords,
                   comesFromPush, isStatement: bool): bool =
   var it = n[i]
-  var key = if it.kind in nkPragmaCallKinds and it.len > 1: it[0] else: it
+  let keyDeep = it.kind in nkPragmaCallKinds and it.len > 1
+  var key = if keyDeep: it[0] else: it
   if key.kind == nkBracketExpr:
     processNote(c, it)
     return
@@ -851,6 +852,9 @@ proc singlePragma(c: PContext, sym: PSym, n: PNode, i: var int,
     inc c.instCounter
     if c.instCounter > 100:
       globalError(c.config, it.info, "recursive dependency: " & userPragma.name.s)
+
+    if keyDeep:
+      localError(c.config, it.info, "user pragma cannot have arguments")
 
     pragma(c, sym, userPragma.ast, validPragmas, isStatement)
     n.sons[i..i] = userPragma.ast.sons # expand user pragma with its content

--- a/tests/pragmas/tuserpragmaargs.nim
+++ b/tests/pragmas/tuserpragmaargs.nim
@@ -1,0 +1,5 @@
+var foo {.exportc: "abc".} = 123
+{.pragma: importc2, importc.}
+var bar {.importc2: "abc".}: int #[tt.Error
+                  ^ user pragma cannot have arguments]#
+echo bar

